xbmc/settings/windows/GUIWindowSettingsCategory.cpp

```diff
@@ -81,12 +82,38 @@ bool CGUIWindowSettingsCategory::OnMessage(CGUIMessage &message)
     case GUI_MSG_WINDOW_INIT:
     {
       m_iSection = (int)message.GetParam2() - (int)CGUIDialogSettingsManagerBase::GetID();
+
+      // Ensure the setting we are about to focus will be visible.
+      auto focusSetting = message.GetStringParam(1);
+      if (!focusSetting.empty())
+      {
+        auto setting = GetSetting(focusSetting);
+        if (setting)
+        {
+          if (CViewStateSettings::GetInstance().GetSettingLevel() < setting->GetLevel())
+            CViewStateSettings::GetInstance().SetSettingLevel(setting->GetLevel());
```

Comment:
you should not do this, because:

1. we do support "lock settings" in profiles (+ master lock enabled). 
2. there are people using kodi in "kiosk mode" with level pre-set to 0 or 1 + skin that does not support swtiching settings levels. by upgrading current settingslevel you effectively broke those.
3. it makes no sense to force show a setting we previously wanted to hide from user, so you should focus it if accessible in current level and noop if not